src/dbus/server/dbus_thread_object.cpp

```diff
@@ -253,6 +253,8 @@ otbrError DBusThreadObject::Init(void)
                                std::bind(&DBusThreadObject::GetUptimeHandler, this, _1));
     RegisterGetPropertyHandler(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_RADIO_COEX_METRICS,
                                std::bind(&DBusThreadObject::GetRadioCoexMetrics, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_BORDER_ROUTING_COUNTERS,
```

Comment:
Should the condition `#if OTBR_ENABLE_BORDER_ROUTING_COUNTERS` be added here or in below implementation?

It seems to me the macro condition is mix-used in either the implementation body or when registering the handler, for instance

https://github.com/openthread/ot-br-posix/blob/6a8db6b66e1b5bb37ef38728bdf27834352367db/src/dbus/server/dbus_thread_object.cpp#L210-L215

Better to unify the style, and I personally prefer a bit to add the macro condition here as it seems useless to register the handler if it turns out to be not enabled.

Reply:
It seems ChannelMonitor is currently the only feature guarded at the declaration. I'd prefer to keep the style consistent with the majority. If we want to migrate to 'guard at declaration' style, we can send a new PR solely for this purpose.

Reply:
Agree that may unify it in a separate PR.

Any preference on the guard style? @abtink @jwhui

Reply:
I agree it makes sense to guard each declaration, but can be done in a separate PR.